slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -105,22 +105,32 @@ ::grpc::Status SLAMServiceImpl::GetMap(ServerContext *context,
     return grpc::Status::OK;
 }
 
-SLAMServiceActionMode SLAMServiceImpl::GetActionMode() {
-    // TODO: Add a case for updating. Requires that an apriori
-    // map is detected and loaded. Will be implemented in this ticket:
-    // https://viam.atlassian.net/browse/DATA-114
-
-    // TODO: Change this depending on the outcome of this scope
-    // doc:
-    // https://docs.google.com/document/d/1RsT-c0QOtkMKa-rwUGY0-emUmKIRSaO3mzT1v6MtFjk/edit#heading=h.tcicyojyqi6c
-    if (map_rate_sec.count() == -1) {
-        LOG(INFO) << "Running in localization only mode";
-        return SLAMServiceActionMode::LOCALIZING;
+void SLAMServiceImpl::DetermineActionMode() {
+    // Check if there is an apriori map in the path_to_map directory
+    std::vector<std::string> map_filenames =
+        viam::io::ListSortedFilesInDirectory(path_to_map);
+
+    // Check if there is a *.pbstream map in the path_to_map directory
+    for (auto filename : map_filenames) {
```

Comment:
[q] the slam directory is very heavily controlled, can we assume that any file in the map directory must be a pbstream?

Reply:
We can not, especially because cartographer runs on osx, as soon as a user takes a look into the folder with data a `.DS_Store` is generated and our code fails.

Reply:
can we just force a deletion of .DS_Store if detected?